src/renderer/components/FtChannelBubble/FtChannelBubble.vue

```diff
@@ -48,13 +48,51 @@
       />
     </div>
     <div
-      :id="sanitizedId"
+      :id="id"
       class="channelName"
     >
       {{ channelName }}
     </div>
   </div>
 </template>
 
-<script src="./ft-channel-bubble.js" />
-<style scoped src="./ft-channel-bubble.css" />
+<script setup>
+import { ref } from 'vue'
+import { useId } from '../../composables/use-id-polyfill'
+
+const id = useId()
+const props = defineProps({
+  channelId: {
+    type: String,
+    required: true
+  },
+  channelName: {
+    type: String,
+    required: true
+  },
+  channelThumbnail: {
+    type: String,
+    default: null
+  },
+  showSelected: {
+    type: Boolean,
+    default: false
+  }
+})
+
+const selected = ref(false)
```

Comment:
As the profile management components access this ref, you need to use defineExpose on it. It's not a problem in Vue 2 as that automatically exposes all variables and functions, but in Vue 3 where the composition API compilation is a lot more optimised, anything that isn't explicitly exposed will be inaccessible from the outside.

I do have a more extensive rewrite in my git stash (I'm sorry I really should have PRed it a while ago) that rewrites the channel bubble and profile management components in the composition API and to use props instead of template refs.

Reply:
Sounds good, I'll close this PR and review yours when it is opened 🙂